src/hatch/utils/env.py

```diff
@@ -1,17 +1,23 @@
+from __future__ import annotations
+
 from ast import literal_eval
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from hatch.utils.platform import Platform
 
 
 class PythonInfo:
-    def __init__(self, platform, executable='python'):
+    def __init__(self, platform: Platform, executable: str = 'python') -> None:
         self.platform = platform
         self.executable = executable
 
-        self.__dep_check_data = None
-        self.__environment = None
-        self.__sys_path = None
+        self.__dep_check_data: dict[str, Any] | None = None
+        self.__environment: dict[str, str] | None = None
```

Comment:
```
❯ python -c "from hatch.utils.env import PythonInfo;from hatch.utils.platform import Platform;print(PythonInfo(Platform()).environment)"
{'implementation_name': 'cpython', 'implementation_version': '3.8.1', 'os_name': 'nt', 'platform_machine': 'AMD64', 'platform_python_implementation': 'CPython', 'platform_release': '10', 'platform_system': 'Windows', 'platform_version': '10.0.19041', 'python_full_version': '3.8.1', 'python_version': '3.8', 'sys_platform': 'win32'}
```

Reply:
I'm not got the point, `_environment` should be `dict[str, str]` based on above output.